internal/commands/app/create_inputs_test.go

```diff
@@ -372,111 +373,659 @@ func TestAppCreateInputsResolveDirectory(t *testing.T) {
+		clusterNames := []string{"Cluster0", "Cluster1"}
+		clusterServiceNames := []string{"mongodb-atlas", "another-data-source"}
+
+		ac := mock.AtlasClient{}
+		ac.ClustersFn = func(groupID string) ([]atlas.Cluster, error) {
+			return []atlas.Cluster{
+				{ID: "456", Name: clusterNames[0]},
+				{ID: "789", Name: clusterNames[1]},
+			}, nil
+		}
+
+		for _, tc := range []struct {
+			description                 string
+			clusterNames                []string
+			clusterServiceNames         []string
+			procedure                   func(c *expect.Console)
+			autoConfirm                 bool
+			expectedClusterServiceNames []string
+		}{
+			{
+				description:                 "use cluster service names provided",
+				clusterNames:                clusterNames,
+				clusterServiceNames:         clusterServiceNames,
+				procedure:                   func(c *expect.Console) {},
+				autoConfirm:                 false,
+				expectedClusterServiceNames: clusterServiceNames,
+			},
+			{
+				description:  "prompt user if no cluster service names are provided",
+				clusterNames: clusterNames,
+				procedure: func(c *expect.Console) {
+					c.ExpectString(fmt.Sprintf("Enter a Service Name for Cluster '%s'", clusterNames[0]))
+					c.SendLine(clusterServiceNames[0])
+					c.ExpectString(fmt.Sprintf("Enter a Service Name for Cluster '%s'", clusterNames[1]))
+					c.SendLine(clusterServiceNames[1])
+					c.ExpectEOF()
+				},
+				autoConfirm:                 false,
+				expectedClusterServiceNames: clusterServiceNames,
+			},
+			{
+				description:         "prompt user if any cluster service name is not provided",
+				clusterNames:        clusterNames,
+				clusterServiceNames: []string{clusterServiceNames[0]},
+				procedure: func(c *expect.Console) {
+					c.ExpectString(fmt.Sprintf("Enter a Service Name for Cluster '%s'", clusterNames[1]))
+					c.SendLine(clusterServiceNames[1])
+					c.ExpectEOF()
+				},
+				autoConfirm:                 false,
+				expectedClusterServiceNames: clusterServiceNames,
+			},
+			{
+				description:                 "default cluster service names to cluster names if not provided and auto confirm is set",
+				clusterNames:                clusterNames,
+				procedure:                   func(c *expect.Console) {},
+				autoConfirm:                 true,
+				expectedClusterServiceNames: clusterNames,
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				console, _, ui, err := mock.NewVT10XConsoleWithOptions(mock.UIOptions{AutoConfirm: tc.autoConfirm})
+				assert.Nil(t, err)
+				defer console.Close()
+
+				doneCh := make(chan (struct{}))
+				go func() {
+					defer close(doneCh)
+					tc.procedure(console)
+				}()
+
+				inputs := createInputs{newAppInputs: newAppInputs{Name: "test-app"},
+					Clusters:            tc.clusterNames,
+					ClusterServiceNames: tc.clusterServiceNames,
+				}
+
+				ds, _, _ := inputs.resolveClusters(ui, ac, "123")
```

Comment:
generally still advisable to handle `err` here even if its expected to be `nil`... just a `assert.Nil(t, err)` on the next line suffices